packages/react/src/components/Footer/LegalNav.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { settings } from 'carbon-components';
+import { Link } from 'carbon-components-react';
+
+const { prefix } = settings;
+
+/**
+ * Footer legal nav component
+ *
+ * @param {object} props react proptypes
+ * @returns {object} JSX object
+ */
+const LegalNav = ({ links }) => {
+  return (
+    <aside
+      data-autoid="footer-legal-nav"
+      className={`${prefix}--legal-nav__container`}>
+      <nav className={`${prefix}--legal-nav`}>
+        <ul className={`${prefix}--legal-nav__list`}>
+          {renderListItems(links)}
+        </ul>
+      </nav>
+    </aside>
+  );
+};
+
+/**
+ * Loops through and renders list items for legal nav
+ *
+ * @param {Array} links A list of links to be rendered
+ * @returns {object} JSX object
+ */
+function renderListItems(links) {
+  const toRender = [];
+
+  for (let i = 0; i < links.length; i += 1) {
```

Comment:
Minor thing and general question to all, wondering if we are ok with going with either a for loop, or perhaps something more terse like map:

```
{links.map((item, index) => (
  <li className={`${prefix}--legal-nav__list-item`}>
    <Link data-autoid="footer-legal-nav__link" href={item.url}>
      {item.title}
    </Link>
  </li>
))}
```

@kennylam @photodow @annawen1 

I wouldn't even mind this within the actual render, or having as a separate function works fine as well. Thoughts?

Reply:
> Minor thing and general question to all, wondering if we are ok with going with either a for loop, or perhaps something more terse like map:

@jeffchew here's my thinking...

`for` loops are hands down more performant, and probably the most versatile. I've been called out on this before for micro performance/premature optimization, and I kind of disagree with that. I feel like you should be building out the cleanest and most performant codebase from the beginning so you don't have to go back and refactor a lot of things later. If you already have maps in there I don't think there is much to gain by replacing them all with for loops unless you know that's a bottleneck to your code, but I see no problem starting with them.

With that said I get `map()` and some of the other loop methods are easier to read, and specific to certain tasks. If that's the direction we want to go I'm cool with that, but to me, the UX and performance that impacts the user, in my opinion, trumps our coding processes/styles in the end.

> I wouldn't even mind this within the actual render or having as a separate function works fine as well. Thoughts?

If you're referring to including it in the JSX render, I'm personally not a fan because it really muddies the JSX up. I don't want to say there is never a scenario where that doesn't make sense, but as a general rule, I personally try to keep logic out of the JSX as much as possible.

Reply:
Here's a pretty detailed article on `map` vs `for`, and I'm a bit in favor of `map`. The article even touches on the performance part of `for`, but seems like it makes more sense and see a difference for super large arrays (which we're not really processing here).

I agree with a separate method for rendering the iteration though.

Thoughts @kennylam @annawen1 ?

Reply:
Yeah, I've seen that article. It's clearly biased. 😜 haha

I don't agree with everything in it, but I think he does bring out some good points on both sides though. I do agree, the difference is definitely shown in larger arrays, and to be fair the gap in performance has been decreasing in browsers.

> However, this is more preference and working in a standard than a clear-cut victory for one or the other.

I think he's right about this, and when it boils down to preference I take the user's side every time. 🤷‍♂️

Reply:
@jeffchew campaigned offline in New York and got the votes for `map()`. So I'll make the updates in this pull request.